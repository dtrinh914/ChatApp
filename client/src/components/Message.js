import React from 'react';

function Message({text}){
    return(
        <li>{text}</li>
    );
}


export default Message;