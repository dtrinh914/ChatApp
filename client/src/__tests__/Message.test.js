import React from 'react';
import Message from '../components/Message';
import {render, getByTestId, cleanup} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

afterEach(cleanup);

//seedData
const testMessage = {
                    "id": "5e28c5c185ffc434b337c266",
                    "text": "This is a test message.",
                    "time": "2020-01-22T22:18:50.097Z",
                    "username": "test_user",
                    "type": "message"
                }

test('should match snapshot', ()=>{
    const {container} = render(<Message message={testMessage} />);
    expect(container).toMatchSnapshot();
});

test('should display username and message', () => {
    const {container} = render(<Message message={testMessage} />);
    const usernameField = getByTestId(container, 'username-time');
    const textField = getByTestId(container, 'text-field');
 
    expect(usernameField).toHaveTextContent('test_user');
    expect(textField).toHaveTextContent('This is a test message.');
});

test('should display the correctly formated date', ()=>{
    const {container} = render(<Message message={testMessage} />);
    const timeField = getByTestId(container, 'username-time');

    expect(timeField).toHaveTextContent('02:18 PM');
});