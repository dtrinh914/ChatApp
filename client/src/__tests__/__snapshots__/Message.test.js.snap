// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <li
    class="MuiListItem-root makeStyles-root-1 MuiListItem-gutters"
    data-testid="message"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-2 MuiAvatar-colorDefault"
    >
      T
    </div>
    <div>
      <p
        class="MuiTypography-root MuiTypography-body1"
        data-testid="username-time"
      >
        <strong>
          test_user
        </strong>
        <span
          class="makeStyles-span-3"
        >
          02:18 PM
        </span>
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1"
        data-testid="text-field"
      >
        This is a test message.
      </p>
    </div>
  </li>
</div>
`;
